[Safe CPP] Address Warnings in rendering/line

stwrt opened the pull request:
#### b8d2efdb55675aefd42c9cf6b22b6ab1126c31a1
<pre>
[Safe CPP] Address Warnings in rendering/line
<a href="https://bugs.webkit.org/show_bug.cgi?id=302112">https://bugs.webkit.org/show_bug.cgi?id=302112</a>
<a href="https://rdar.apple.com/problem/164194446">rdar://problem/164194446</a>

Reviewed by Chris Dumez.

Fix Safe CPP warnings rendering/line/

rendering/line/LineBreaker.h
rendering/line/LineWidth.h
rendering/line/TrailingObjects.h
rendering/line/WordTrailingSpace.h

*  Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations
* Source/WebCore/SaferCPPExpectations/NoUncheckedPtrMemberCheckerExpectations
* Source/WebCore/rendering/line/LineBreaker.h
* Source/WebCore/rendering/line/LineWidth.cpp
* Source/WebCore/rendering/line/LineWidth.h
* Source/WebCore/rendering/line/TrailingObjects.cpp
* Source/WebCore/rendering/line/TrailingObjects.h
* Source/WebCore/rendering/line/WordTrailingSpace.h

Canonical link: <a href="https://commits.webkit.org/302726@main">https://commits.webkit.org/302726@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/596fd43de5ca20fa4a7530e79ea836b3a8f0cc79

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129993 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2254 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40854 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137388 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81497 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/1d1d92a6-a9e4-4067-9917-3db0b786e41c) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131864 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2216 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2146 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/99018 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/66819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/407297ce-6ad5-482f-8fb6-0899d742f8f9) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132940 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1647 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116426 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79716 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2dfa8334-b4b7-4973-92f5-6169639db031) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1563 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34551 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80657 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110084 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35059 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139867 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/2048 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1891 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107524 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2093 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112773 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107415 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27347 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1631 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31233 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/54925 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2121 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/65490 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1936 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1970 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/2044 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/rendering/line/LineWidth.h:
```diff
@@ -76,7 +77,7 @@ class LineWidth {
     void updateLineDimension(LayoutUnit newLineTop, LayoutUnit newLineWidth, float newLineLeft, float newLineRight);
 
-    RenderBlockFlow& m_block;
+    CheckedPtr<RenderBlockFlow> m_block;
```
Since it used to be a reference, it should be a CheckedRef, not a CheckedPtr. Also, we should make it `const` so we don't need variables on the stack for it.

cdumez commented on Source/WebCore/rendering/line/WordTrailingSpace.h:
```diff
@@ -45,15 +46,16 @@ struct WordTrailingSpace {
 private:
     enum class WordTrailingSpaceState { Uninitialized, Initialized };
-    const RenderStyle& m_style;
+    CheckedRef<const RenderStyle> m_style;
```
We should mark this data member as `const`, then you'll no longer need `style` stack variables.

cdumez commented on Source/WebCore/rendering/line/WordTrailingSpace.h:
```diff
@@ -45,15 +46,16 @@ struct WordTrailingSpace {
 
-        auto& font = m_style.fontCascade();
-        m_width = font.width(RenderBlock::constructTextRun(span(space), m_style), &fallbackFonts) + font.wordSpacing();
+        CheckedRef style = m_style.get();
```
Not needed if we make data member `const`.

cdumez reviewed (approved):
LGTM

stwrt committed:
[Safe CPP] Address Warnings in rendering/line
https://bugs.webkit.org/show_bug.cgi?id=302112
rdar://problem/164194446

Reviewed by Chris Dumez.

Fix Safe CPP warnings rendering/line/

rendering/line/LineBreaker.h
rendering/line/LineWidth.h
rendering/line/TrailingObjects.h
rendering/line/WordTrailingSpace.h

*  Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations
* Source/WebCore/SaferCPPExpectations/NoUncheckedPtrMemberCheckerExpectations
* Source/WebCore/rendering/line/LineBreaker.h
* Source/WebCore/rendering/line/LineWidth.cpp
* Source/WebCore/rendering/line/LineWidth.h
* Source/WebCore/rendering/line/TrailingObjects.cpp
* Source/WebCore/rendering/line/TrailingObjects.h
* Source/WebCore/rendering/line/WordTrailingSpace.h

Canonical link: https://commits.webkit.org/302726@main
